src/meltano/core/tracking/contexts/environment.py

```diff
@@ -44,9 +44,19 @@ def _get_parent_context_uuid_str() -> str | None:
 class EnvironmentContext(SelfDescribingJson):
     """Environment context for the Snowplow tracker."""
 
+    ci_markers = {"GITHUB_ACTIONS", "CI"}
+    notable_flag_env_vars = {"CODESPACES", *ci_markers}
+
+    @classmethod
+    def _true_notable_flag_env_vars(cls) -> Iterable[str]:
+        for env_var_name in cls.notable_flag_env_vars:
+            env_var_value = os.environ.get(env_var_name, "0")
+            with suppress(ValueError):
+                if strtobool(env_var_value):
+                    yield env_var_name, env_var_value
```

Comment:
So we're only checking _if_ the value is truthy, and then only send the raw string value to snowplow?

Reply:
f520ad8 makes it so that all *set* env vars listed are collected. Their values can be `true`, `false`, or `null`, depending on if they're truthy, falsey, or something else respectively.